korge-spine/src/commonTest/kotlin/com/esotericsoftware/spine/BonePositionTest.kt

```diff
@@ -0,0 +1,57 @@
+package com.esotericsoftware.spine
+
+import com.esotericsoftware.spine.Animation.MixBlend
+import com.esotericsoftware.spine.Animation.MixDirection
+import com.esotericsoftware.spine.attachments.*
+import com.soywiz.korio.async.suspendTest
+import com.soywiz.korio.file.std.resourcesVfs
+import com.soywiz.korio.util.OS
+import kotlin.test.Test
+
+
+class BonePositionTest {
+    @Test
+    fun test() = suspendTest({ !OS.isJs }) {
+        val resource = resourcesVfs["spineboy/spineboy-ess.json"]
+        val json = SkeletonJson(AttachmentLoaderTest())
+        val skeletonData = json.readSkeletonData(resource)
+        val skeleton = Skeleton(skeletonData)
+        val bone = skeleton.findBone("gun-tip")
+        val fps = 1 / 15f
+        for (animation in skeletonData.animations) {
+            var time = 0f
+            while (time < animation.duration) {
+                animation.apply(skeleton, time, time, false, null, 1f, MixBlend.first, MixDirection.`in`)
+                skeleton.updateWorldTransform()
+                println(animation.name + "," + bone!!.worldX + "," + bone.worldY + "," + bone.worldRotationX)
+                time += fps
+            }
+        }
+    }
+}
+
+class AttachmentLoaderTest: AttachmentLoader {
```

Comment:
To not require this class, maybe you could go to the AttachmentLoader interface and mark all the methods with a default implementation, or even making it an open class:

```
interface AttachmentLoader {
    fun newRegionAttachment(skin: Skin, name: String, path: String): RegionAttachment?
    // ...
}
```
->
```
interface AttachmentLoader {
    fun newRegionAttachment(skin: Skin, name: String, path: String): RegionAttachment? = null
    // ...
}
```
or
```
open class AttachmentLoader {
    open fun newRegionAttachment(skin: Skin, name: String, path: String): RegionAttachment? = null
    // ...
}
```

After that,
```
val json = SkeletonJson(AttachmentLoaderTest())
```
->
```
val json = SkeletonJson(object : AttachmentLoader { }) // interface version
val json = SkeletonJson(object : AttachmentLoader() { }) // class version
```

Reply:
I'll go for the interface version